packages/@glimmer/compiler/lib/passes/1-normalization/keywords/utils/curry.ts

```diff
@@ -0,0 +1,56 @@
+import { ASTv2, generateSyntaxError } from '@glimmer/syntax';
+
+import { Err, Ok, Result } from '../../../../shared/result';
+import { NormalizationState } from '../../context';
+
+export function assertValidCurryUsage(type: string, stringsAllowed: boolean) {
+  return (
+    node: ASTv2.AppendContent | ASTv2.InvokeBlock | ASTv2.CallExpression,
+    state: NormalizationState
+  ): Result<{
+    definition: ASTv2.ExpressionNode;
+    args: ASTv2.Args;
+  }> => {
+    let { args } = node;
+
+    let definition = args.nth(0);
+
+    if (definition === null) {
+      return Err(
+        generateSyntaxError(
+          `${type} requires a component definition or identifier as its first positional parameter, did not receive any parameters.`,
```

Comment:
I think the message here is a bit wrong for helpers. The phrasing "component definition" doesn't make sense in a helper context.

Reply:
ahh, yes, definitely need to update that